config/stp.py

```diff
@@ -319,6 +423,76 @@ def get_global_stp_priority(db):
     return priority
 
 
+def get_bridge_mac_address(db):
+    """Retrieve the bridge MAC address from the CONFIG_DB"""
+    device_metadata = db.get_entry('DEVICE_METADATA', 'localhost')
+    bridge_mac_address = device_metadata.get('mac')
+    return bridge_mac_address
+
+
+def do_vlan_to_instance0(db):
+    """Get VLAN list and create MST instance 0"""
+    vlan_list = db.get_table('VLAN').keys()
+    if vlan_list:
+        vlan_list_str = ','.join(vlan_list)
+        mst_inst_fvs = {
+            'bridge_priority': MST_DEFAULT_BRIDGE_PRIORITY,
+            'vlan_list': vlan_list_str
+        }
+        db.set_entry('STP_MST', 'MST_INSTANCE|0', mst_inst_fvs) #Is it STP_MST_INST or STP_MST?
+
+def enable_mst_for_interfaces(db):
+    fvs = {
+        'enabled': 'true',
+        'root_guard': 'false',
+        'bpdu_guard': 'false',
+        'bpdu_guard_do_disable': 'false',
+        'portfast': 'false',
+        'uplink_fast': 'false',
+        'edge_port': 'false',
+        'link_type': MST_AUTO_LINK_TYPE,
+        'path_cost': MST_DEFAULT_PORT_PATH_COST,
+        'priority': MST_DEFAULT_PORT_PRIORITY
+        }
+    port_dict = natsorted(db.get_table('PORT'))
+    intf_list_in_vlan_member_table = get_intf_list_in_vlan_member_table(db)
+
+    for port_key in port_dict:
+        if port_key in intf_list_in_vlan_member_table:
+            db.set_entry('STP_MST_PORT', f"MST_INSTANCE|0|{port_key}", fvs)
+
+    po_ch_dict = natsorted(db.get_table('PORTCHANNEL'))
+    for po_ch_key in po_ch_dict:
+        if po_ch_key in intf_list_in_vlan_member_table:
+            db.set_entry('STP_MST_PORT', f"MST_INSTANCE|0|{po_ch_key}", fvs)
+
+
+def disable_global_pvst(db):
+    db.set_entry('STP', "GLOBAL", None)
+    db.delete_table('STP_VLAN')
+    db.delete_table('STP_PORT')
+    db.delete_table('STP_VLAN_PORT')
+
+def disable_global_mst(db):
+    db.set_entry('STP', "GLOBAL", None)
+    db.delete_table('STP_MST')
+    db.delete_table('STP_MST_INST')
+    db.delete_table('STP_MST_PORT')
+    db.delete_table('STP_PORT')
+
+
+def update_mst_instance_parameters(ctx, db, param_type, new_value):
+    """Update MST instance parameters in the STP_MST_INST table"""
+
+    allowed_params = {"max_hops", "max_age", "hello_time", "forward_delay"}
+    if param_type not in allowed_params:
+        ctx.fail("Invalid parameter")
+
+    mst_inst_table = db.get_table('STP_MST_INST')
```

Comment:
This value should be added in STP_MST table (Global). 
MSTP has only one timer which is global

Reply:
I have addressed this comment, please review